scala/private/phases/phase_collect_jars.bzl

```diff
@@ -0,0 +1,71 @@
+#
+# PHASE: collect jars
+#
+# DOCUMENT THIS
+#
+load(
+    "@io_bazel_rules_scala//scala/private:rule_impls.bzl",
+    "collect_jars_from_common_ctx",
+)
+
+def phase_test_collect_jars(ctx, p):
```

Comment:
I used the convention like
`scala_test` -> `phase_test_collect_jars`
`scala_repl` -> `phase_repl_collect_jars`

If we make it `phase_scala_test_collect_jars`, should we also make it `phase_scala_repl_collect_jars`? I feel like it will significantly increase the phase name but all rules start with `scala_`. Why do you think it's ambiguous?

Or we should come up with a better naming convention?

Reply:
scala_test is indeed a problematic one, it should probably have been scalatest_test but it was the first one so it grabbed the ambigous name. I think repl can stay as is but I thought that this phase is general for all tests while it's not so for junit scala tests right? This is what I think we should namespace it. Maybe `phase_scalatest_collect_jars`?

Reply:
I think these other 4 phases also fall into this category
`phase_test_compile`
`phase_test_final`
`phase_test_runfiles`
`phase_test_write_executable`

Should we also change them from `_test_` to `_scalatest_`?

Reply:
No worries. Fixed.